### PR TITLE
Sfu Cog Fixes

### PR DESCRIPTION
**Pull request by TitanVJ**
# SFU Cog Edge Cases, Fixes, and slight improvements 

## Get request fixes
SFU api sometimes will return data as a whole or chunked. Requests could handle either no problem with its json() method but aiohttp's json() method couldnt. So I switched the `outline` and `sfu` commands to read in the data as if it were chunked; So it reads in 10 bytes at a time and decodes them and loads the json data.

## API URL Keyword problem
Mix up of key words `registration` and `current` caused some data to be pulled from the wrong semester

## Next functionality in outline command
The keyword fix highlighted this idea and so if the keyword `next` is included in the users args it'll look to the next semester. Logic in the command has been updated to work with this functionality. The command will return and error if `next` is used when it is not registration time, since the outlines won't exist.

### Updated all the things like test cases and help.json

### CODE REVIEW

**Comment by TitanVJ** [comment]
also wrapped a few things in try/except since the api doesnt always include those json keys. and i added a co-req field cuz i forgot those exist lol

File: commands_to_load/help.json [hunk 1]
```diff
@@ -1,46 +1,46 @@
-{ "commands" : [
```
Comment (modernNeo) [inline commands_to_load/help.json]: you messed up with help.json
please do 
`git fetch`
`git checkout origin/master -- commands_to_load/help.json`

after doing that, add the command you want to add, but there has been a convention change in the file. if you may notice, the value for `access` has been changed.

These now map to either `role` or `channel.` What this basically means is if the access to that command is based on the user's `role`, then the value is `role` the there is an additional key-value pair like so `( "role": "<what _role>")`
There is also another possibility. If the access is based on the user's permission in the `channel` that they called the command, then the value is `channel` for key `access` and there is an additional key-value pair like so `("channel":"<what_permission_in_channel_is_needed_for_access")`

Comment (TitanVJ) [reply commands_to_load/help.json]: done

**Comment by wnfldchen** [comment]
Interesting changes; looking at

**Comment by wnfldchen** [comment]
I know the official example uses `await resp.content.read(10)` and it works, but it might not be the best way of going about it

**Comment by TitanVJ** [comment]
ill work on this on monday cuz i have to spent the weekend studying for linalg mt on monday

**Comment by wnfldchen** [comment]
Awesome, good luck! In the meantime I'll be doing the same async networking changes ro the urban command too in my own branch so feel free to check out those upcoming changes for inspiration too

**Comment by wnfldchen** [comment]
Actually, aiohttp has its own json decoder; is there a need to go chunk by chunk at all?

**Comment by TitanVJ** [comment]
@wnfldchen original switch over to aiohttp used the json decoder but its get stucked when it encounters the chunked responses so had to switch over to this current method. i meant to point this out in the pr description(i've edited now, so it does)

**Comment by wnfldchen** [comment]
im doing it this way and aiohttp's json decoder works: https://github.com/CSSS/wall_e/blob/6f8d766feb4d1ba50ed3d8d434d63f2a4e6ce110/commands_to_load/Misc.py#L86 did you do it like this? (`await` the call to json decoder, and place everything in the `async with`)

**Comment by TitanVJ** [comment]
https://github.com/CSSS/wall_e/blob/master/commands_to_load/SFU.py#L176

**Comment by TitanVJ** [comment]
i tested it locally nesting in the async with and still behaved the same

**Comment by modernNeo** [comment]
@TitanVJ friendly hint when linking code in PRs: https://help.github.com/articles/creating-a-permanent-link-to-a-code-snippet/

**Comment by TitanVJ** [comment]
@wnfldchen done fixing the things you requested. have at it :)

**Pull request merged**